pkg/kubelet/server/server.go

```diff
@@ -240,11 +253,27 @@ func NewServer(
 	resourceAnalyzer stats.ResourceAnalyzer,
 	auth AuthInterface,
 	kubeCfg *kubeletconfiginternal.KubeletConfiguration) Server {
+	var tp oteltrace.TracerProvider
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletTracing) {
+		opts := []otlpgrpc.Option{}
+		if kubeCfg.Tracing.Endpoint == nil {
+			opts = append(opts, otlpgrpc.WithEndpoint(*kubeCfg.Tracing.Endpoint))
+		}
+		opts = append(opts, otlpgrpc.WithInsecure())
+		var err error
+		tp, err = traces.InitTracing(context.Background(), host.GetHostname(), kubeletServiceName, kubeCfg.Tracing.SamplingRatePerMillion, opts)
```

Comment:
If possible, we should re-use the tracer-provider created in cmd/kubelet/app.  kubedeps is probably the right place to store it.

Reply:
ok, got it. I added to kubeDeps